### PR TITLE
feat/wireguard customizable service type ingresses annotations available

### PR DESCRIPTION
**Pull request by mateusclira**
- feat: creating annotations for ui and rest
- feat: wireguard can have annotations and can change servicetype
- refactor: updating ingress and wireguard for accepting their file changes

### CODE REVIEW

File: templates/ingress.yaml [hunk 1]
```diff
@@ -38,6 +38,7 @@ metadata:
     {{- else if .tls.enabled}}
     cert-manager.io/cluster-issuer: {{ .tls.issuerName }}
     {{- end }}
+    {{ (((.Values).ingress).annotations).ui }}
```
Comment (jfmrm) [inline templates/ingress.yaml:41]:
```suggestion
    {{ .annotations.ui }}
```

File: templates/ingress.yaml [hunk 2]
```diff
@@ -96,6 +97,7 @@ metadata:
     {{- else if .tls.enabled}}
     cert-manager.io/cluster-issuer: {{ .tls.issuerName }}
     {{- end }}
+    {{ (((.Values).ingress).annotations).rest }}
```
Comment (jfmrm) [inline templates/ingress.yaml:100]:
```suggestion
    {{ .annotations.rest }}
```

File: templates/services.yaml [hunk 3]
```diff
@@ -37,9 +37,13 @@ metadata:
...
     {{- include "netmaker.labels" . | nindent 4 }}
   name: '{{ include "netmaker.fullname" . }}-wireguard'
+  {{- with .Values.service.annotations }}
+  annotations:
+    {{- toYaml .Values.service.annotations | nindent 4 }}
+  {{- end }}
 spec:
   externalTrafficPolicy: Local
-  type: NodePort
+  type: {{ (((.Values).wireguard).service).servicetype }}
```
Comment (jfmrm) [inline templates/services.yaml:46]:
```suggestion
  type: {{ .Values.wireguard.service.serviceType }}
```

File: values.yaml [hunk 4]
```diff
@@ -99,6 +101,9 @@ ingress:
 wireguard:
   # -- whether or not to use WireGuard on server
   enabled: true
+  annotations: {}
+  service:
+    servicetype: NodePort
```
Comment (jfmrm) [inline values.yaml:106]:
```suggestion
    serviceType: NodePort
```

**Review by jfmrm — CHANGES_REQUESTED** [review]
Looks good, just a few style changes

**Comment by mateusclira** [comment]
Substituted for https://github.com/gravitl/netmaker-helm/pull/19

**Pull request closed without merging**